Drive/wheel.hpp

```diff
@@ -23,68 +23,73 @@ class Wheel
     steer_motor_.Enable(enable);
   };
 
-  /// Return the speed of the hub motor in RPM.
-  /// Look into rmd_x function RequestFeedbackFromMotor() & GetFeedback()
-  units::angular_velocity::revolutions_per_minute_t GetSpeed()
+  /// Gets the speed of the hub motor.
+  double GetSpeed()
   {
-    return hub_speed_;
+    return hub_speed_.to<double>();
   };
 
-  /// Return the angle/position of the steering motor.
-  /// Look into rmd_x function RequestFeedbackFromMotor() & GetFeedback()
-  units::angle::degree_t GetPosition()
+  /// Gets the angle/position of the steering motor.
+  double GetPosition()
   {
-    return zero_offset_angle_;
+    return homing_offset_angle_.to<double>();
   };
 
-  /// Sets the speed of the hub motor and updates hub_motor_ variable so long as
-  /// desired speed is under the max speed
+  /// Sets the speed of the hub motor. Will not surpass max/min value
+  /// @param hub_speed the new speed of the wheel
   void SetSpeed(units::angular_velocity::revolutions_per_minute_t hub_speed)
   {
     if (hub_speed > kMaxHubSpeed)  // need use abs(hub_speed) ?
     {
       hub_speed = kMaxHubSpeed;
     }
+    if (hub_speed < kMaxNegativeHubSpeed)  // need use abs(hub_speed) ?
+    {
+      hub_speed = kMaxNegativeHubSpeed;
+    }
     hub_speed_ = hub_speed;
     steer_motor_.SetSpeed(hub_speed);
```

Comment:
Wouldn't you want to change the hub motor speed and not the steering motor speed? Where do you change the speed of the hub motor so that the wheels go forward. Wouldn't the steering motor just be used to change the angle of the motor?

Reply:
yeah lol good catch